packages/itwinui-react/src/core/utils/hooks/useStyles.ts

```diff
@@ -11,6 +11,14 @@ import * as React from 'react';
 // @ts-ignore
 import rawCssText from '../../../styles.js';
 
+/** rule that reverts all v1 styles when used inside a v2 boundary */
+const revertV1Styles = (() => {
+  // using :not(#\#) to artifically inflate the specificity of * selector
+  const allSelector = `:where(.iui-body, [class*='iui-theme-']) :where(.iui-root) :not(#\\#)`;
+  const revertRule = `${allSelector}, ${allSelector}::before, ${allSelector}::after { all: revert-layer; }`;
```

Comment:
Seems like revert-layer is not widely supported yet. https://developer.mozilla.org/en-US/docs/Web/CSS/revert-layer#browser_compatibility
I think we need more testing before adding this.

Reply:
the data is wrong. it is widely supported everywhere that cascade layers are supported.
https://wpt.fyi/results/css/css-cascade?label=master&label=stable&product=chrome&product=firefox&product=safari&aligned&q=revert-layer